### PR TITLE
[MAINT] run tests in parallel

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes none

Changes proposed in this pull request:

- run tests in parallel
- reduce size of the data for some decoding tests

### CODE REVIEW

**Comment by Remi-Gau** [comment]
test that hangs on windows with python 3.12:

nilearn/decoding/tests/test_decoder.py::test_decoder_regression_screening[svr-20]

probably worth checking why this is the case

**Comment by Remi-Gau** [comment]
@bthirion @man-shu 
Would not mind a review on this as it definitely speeds up the test suite: runs now take between 10 and 25 minutes (as opposed to the 25 to 50 minutes currently).

File: pyproject.toml [hunk 1]
```diff
@@ -81,8 +81,9 @@ test = [
     "coverage",
     "pytest>=6.0.0",
     "pytest-cov",
-    "pytest-random-order",
+    "pytest-randomly",
```
Comment (man-shu) [inline pyproject.toml:84]: Just curious, what does this do differently from `pytest-random-order` that improves the speed?

I read this: https://pytest-dev.github.io/pytest-order/stable/other_plugins.html, but it doesn't seem to have an explanation for the speed improvement.

**Review by man-shu — APPROVED** [review]
LGTM!

File: pyproject.toml (hunk 1 above)
Comment (Remi-Gau) [reply pyproject.toml:84]: ah no xdist is the pytest plugin that parallelize but pytest-random-order is not compatible with it so I had to pick another plugin to randomize the test order

**Review by bthirion — APPROVED** [review]
LGTM, thx.

**Pull request merged**